### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by ridaarif98**
-  Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
-  Set up associations between models.

My models have included some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method that returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.
- ApplicationController
   -  Create a method called current_user in my ApplicationController.
    -  A current_user method returns the first user from a database.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] As per the given _ERD_, the _Comment_ **association** should have _Comment_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (rloterh) [inline app/models/like.rb]: - [x] As per the given _ERD_, the _Like_ **association** should have _Like_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :user
```
Comment (rloterh) [inline app/models/post.rb]: - [x] As per the given _ERD_, the _Post_ **association** should have _Post_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  # Removes one or more objects from the collection by setting their foreign keys to NULL.
+  # Objects will be in addition destroyed if they're associated with dependent: :destroy
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
```
Comment (rloterh) [inline app/models/user.rb]: - [x] As per the given _ERD_, the _User_ **association** should have _author_id_ referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve this aspect, as can be seen below:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Rida, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ridaarif98** [comment]
@rloterh thank you so much. I will fix them

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Rida, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ridaarif98** [comment]
@rloterh thank you

**Pull request merged**